### DISCUSSION TITLE
Vertical-Tail generates Mach correction error

### DISCUSSION THREAD

**Discussion by vincinu** [byuflowlab/FLOWUnsteady#92] [Q&A]
Hi all, I am trying to run a simulation for a small commuter aircraft. To start, I was able to input a simple geometry including fuselage, wing and Horizontal-Tail. However, when I try to add a VTP, the simulation breaks, giving the Mach correction error (Mach>1). In particular, this happens as soon as I try to rotate the VTP (of 90 deg) to its vertical position using the following lines of code:

vtp = vlm.complexWing(b_vtp*2, ar_vtp, n_wing, [0.0,1.0], [1.0/tr_vtp,1.0], ones(2)*twist_root_vtp, 
						ones(1)lambda_vtp, ones(1)*gamma_vtp, symmetric=false);

O_vtp = [dx_vtp, 0.0, dz_vtp]                                
Oaxis_vtp = uns.gt.rotation_matrix2(90.0, 0.0, 0.0)  
vlm.setcoordsystem(vtp, O_vtp, Oaxis_vtp)

Do you have any advice on how to overcome this issue? Or any ideas about possible causes?

Any help is much appreciated! Thank you all.

**Comment by EdoAlvarezR** [comment]
Hi Vincinu! Have you tried visualizing the simulation? A lot of issues can be easily spotted with visualization (e.g., blades overlapping, rotating in the wrong direction, etc). I'd recommend following this tutorial: [https://flow.byu.edu/FLOWUnsteady/visualization/](https://flow.byu.edu/FLOWUnsteady/visualization)

**Reply by vincinu** [reply]
Hi Edo, thank you for your answer. Unfortunately, the only thing I noticed in the visualization is that the VTP was miss-sized due to the "symmetric=false" constraint. After correcting it, I keep bumping in the same error. When running the simulation without rotating the VTP wing element or excluding the VTP completely, it works without issues. I think the error might be related to the VTP and the way I am rotating the wing element (VTP). I also tried doing the same with the function "uns.gt.rotation_matrix" instead of "uns.gt.rotation_matrix2", but no luck with that either. Still looking into it. Thank you again for your help.
